src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,1041 @@
+            visibleDependency: 'showNumbers',
+            visiblePredicate: (dependentValue: boolean) =>
+                dependentValue === true,
+        },
+        /** md
+### Background Color: Background color of the grid
+         **/
+        backgroundColor: {
+            value: this.backgroundColor,
+            forceType: 'color',
+            displayName: 'Background color',
+            required: false,
+        },
+    }
+
+    constructor() {
+        super()
+        /** md
+### Property 1, 2, etc.:  Properties to display by coloring cells
+
+You can add multiple properties.  For each, there are some parameters
+to set.
+
+##### Property:  the property to highlight
+
+The description for each property specifies the conditions under which
+that property holds for a given integer.
+
+- None:  This is simply a placeholder to indicate that no further properties
+will be used.  Choosing anything other than none will add a new property
+and reveal parameters for it.
+- Prime:  Its absolute value is prime
+- Negative:  Less than zero
+- Even:  Divisible by two
+- Odd: Not even
+- Divisible_By:  Divisible by the specified divisor
+- Last_Digit_Is:  The final digit base 10 is the specified digit
+- Polygonal_Number:  Positive and that many dots can be arranged in a
+  polygonal arrangement with the specified number of sides.
+- Sum_Of_Two_Squares:  Nonnegative and equal to the sum of two squares
+- Abundant:  Its absolute value exceeds the sum of its proper divisors
+- Perfect:  Equal to the sum of its proper divisors
+- Deficient:  Its absolute value is less than the sum of its proper divisors
+- Semi_Prime:  Its absolute value is a
+  [semi-prime](https://en.wikipedia.org/wiki/Semiprime), that is, a product of
+  exactly two primes (possibly equal)
+
+##### Display:  Highlight style for cells with the property
+
+Using different display styles allows for visualizing two properties
+that are both true for the same value at once,
+without them overcoloring each other. (Note that later properties overcolor
+earlier ones that use the _same_ style.)
+
+- Fill_Cell:  Fill the complete cell
+- Box_In_Cell:  Fill only a smaller central box in the cell
+
+##### Color:  Highlight color for cells with the property
+         **/
+        for (let i = 0; i < MAXIMUM_ALLOWED_PROPERTIES; i++) {
+            const ithPropertyObject = {
+                property: i === 0 ? Property.Prime : Property.None,
+                visualization:
+                    i === 1
+                        ? PropertyVisualization.Box_In_Cell
+                        : PropertyVisualization.Fill_Cell,
+                color: DEFAULT_COLORS[i],
+                aux: 3n,
+            }
+            this.propertyObjects.push(ithPropertyObject)
+            Object.assign(
+                this.params,
+                getPropertyParams(i, ithPropertyObject)
+            )
+        }
+    }
+
+    checkParameters() {
+        const status = super.checkParameters()
```

Comment:
I think we can just return `super.checkParameters()`, which I will do.

Reply:
Done in https://github.com/numberscope/frontscope/pull/230/commits/ac7ac7363ad696a4c17ea965e5f52b43c836f09b.

Reply:
Also looks good